### PR TITLE
Changes for LEDE/WRT integration

### PR DESCRIPTION
**Pull request by skids**
Allow a packaging system to alter compiled in defaults, and
allow changing the control plane net/mask from the CLI.
Also freshen up README.md including fixing some rules

LEDE packaging files are available in skids/lede-custom-repo.

### CODE REVIEW

**Comment by skids** [comment]
Very nice implementation.  I successfully tested packages generated (via the
other repo mentioned) on a LEDE device for both mysql and pgsql.  Let me know
how you want to arrange things as far as keeping a separate fork for LEDE integration
or having users of those systems pull directly from your upstream.

**Comment by michael-dev** [comment]
Nice to hear. I'm happy to include the changes needed for LEDE packaging into master. I'll need to look a bit deeper into the changes related to cli as I'm currently wondering how the defaults get processed if the argument is not provided.

**Comment by skids** [comment]
Ah!  Good point.  Those should not be required_argument and should DTRT if not present.  I'll see about adding another commit for that.

**Comment by skids** [comment]
OK, that should fix it up, other than no log messages without the options being supplied.

I have to decide whether it is better to point the LEDE custom feed repo directly at your repo, or keep the fork as the source.  I suppose that that can be changed back and forth anytime depending on need.  I could give you write access to the LEDE custom feed repo in case I get hit by a truck and a security issue comes along, if you want.

Also, while I have your attention, it doesn't look to me that the sample rules manage to catch DHCP RELEASE packets... could you review them against your production rules?

File: src/udp.c [hunk 1]
```diff
@@ -178,14 +273,20 @@ void udp_receive(int udpsocket, void* ctx)
 
 static __attribute__((constructor)) void udp_init()
```
Comment (michael-dev) [inline src/udp.c]: There is a problem with udp_init. Calling add_option_cb does not enforce the set_* methods to be run instantly, quite the opposite, they will only run once parse_cmdline is called from main.c. So dhcpsnoopingd will not listen on the newly configured port but instead udp_init will listen on the default port.

**Comment by michael-dev** [comment]
> I have to decide whether it is better to point the LEDE custom feed repo directly at your repo, or keep
> the fork as the source. I suppose that that can be changed back and forth anytime depending on need.
> I could give you write access to the LEDE custom feed repo in case I get hit by a truck and a security
> issue comes along, if you want.

I have no strong opinion on this, I'm fine with both ways.

> Also, while I have your attention, it doesn't look to me that the sample rules manage to catch DHCP
> RELEASE packets... could you review them against your production rules?

The example filters catch all UDP/IPv4 traffic on port 67/68 with the wifi client sending on port 68 and the dhcp server using port 67. Why should this rule not catch DHCPRELEASE?

**Comment by michael-dev** [comment]
Maybe we could defer the code to open the udp socket into a zero-delay timer using cb_add_timer, so it will be run after all arguments have been processed.

**Pull request merged**